src/python/graplctl/graplctl/common.py

```diff
@@ -110,17 +110,46 @@ def get_command_results(
             break
 
     for instance_id in instance_ids:
-        invocation = ssm.get_command_invocation(
-            CommandId=command_id,
-            InstanceId=instance_id,
-            PluginName="runShellScript",
+        invocation = get_command_invocation(
+            ssm=ssm,
+            command_id=command_id,
+            instance_id=instance_id,
         )
 
         if invocation["Status"] == "Success":
             yield instance_id, invocation["StandardOutputContent"].strip()
         else:
             raise SSMException(invocation)
 
+def get_command_invocation(
+    ssm: SSMClient,
+    command_id: str,
+    instance_id: str,
+) -> GetCommandInvocationResultTypeDef:
+    """
+    get-command-invocation commonly throws an awfully named exception: InvalidPluginName.
+    The actual meaning? This invocation hasn't become available.
+    This wrapper makes it a bit more sane to use.
+    """
+    LOGGER.info(
+        f"retrieving invocation metadata for command {command_id} on instance {instance_id}"
+    )
+    while True:
```

Comment:
Should there be some kind of maximum amount of time to wait before finally erroring out?